src/appengine/handlers/cron/throttler.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Bug throttling."""
+
+import datetime
+
+from clusterfuzz._internal.config import local_config
+from clusterfuzz._internal.datastore import data_handler
+from clusterfuzz._internal.datastore import data_types
+from clusterfuzz._internal.metrics import logs
+
+MAX_BUGS_PER_PROJECT_PER_24HRS_DEFAULT = 100
+
+
+class Throttler:
+  """Bug throttler"""
+
+  def __init__(self):
+    self.bug_filed_per_job_per_24hrs = {}
```

Comment:
nit: make all of these "private" by prefixing the members with "_".

e.g. self._bug_filed_per_job_per_24hrs